### ISSUE TITLE
TODO lints incorrectly include dartdoc if placed right before one.

### ISSUE THREAD

**Issue by rrousselGit** [dart-lang/linter#4829]
Hello!

TL;DR when we write:

```dart
// TODO: ...
/// Some Dartdoc
void fn() {}
```

The warning triggered by the TODO also highlights the entire Dartdoc, instead of stopping at the first line:

<img width="177" alt="Screenshot 2023-12-28 at 16 34 18" src="https://github.com/dart-lang/linter/assets/20165741/63a6551e-e6ba-4adb-b244-0cd2ed3641da">

**Comment by parlough** [comment]
I can't seem to move issues from this repository, but I believe the TODO code is implemented as a special diagnostic type directly in the analysis server.

However for this case, I do wonder if it's caused by special handling in the VS Code plugin since I thought the analysis server explicitly checked if the TODO continued on a line that is also a single line comment.

\cc @DanTup

**Comment by pq** [comment]
Hi @rrousselGit! This isn't reproducing for me in IDEA or VS Code. What version of dart are you using?

Testing on: 

    Dart SDK version: 3.3.0-edge.4e6491c8d3dedceb4696c331d66c129340ac3a16 (main) (Tue Nov 14 19:46:59 2023 +0000)


(I'll update to latest and try again.)

**Comment by rrousselGit** [comment]
I'm using:

```
Dart SDK version: 3.3.0-256.0.dev (dev) (Thu Dec 21 20:03:06 2023 -0800) on "macos_arm64"
```

As for VScode:

```
1.85.1
0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2
arm64
```

And the extensions:

```
Dart-Code.dart-code@3.80.0
Dart-Code.flutter@3.81.20231227
```

**Comment by srawlins** [comment]
What are "TODO lints"?

Anyhow, I also cannot reproduce this, even with `"dart.showTodos": false,` commented out in my VS code settings. Nothing I do shows blue squiggles across both lines.

VS Code 1.85.1
Dart Code 3.80.0.
Dart at SDK HEAD-ish

**Comment by rrousselGit** [comment]
If that helps, it's worth noting that it's picked up by the TODO's description too:

<img width="250" alt="Screenshot 2023-12-28 at 21 07 17" src="https://github.com/dart-lang/linter/assets/20165741/abf94654-efc5-4153-ba58-c9f2f180d89f">


And this only happens if following `//` with `///`. Using `//` and `//` or `///` and `///` does not cause the problem:

<img width="122" alt="Screenshot 2023-12-28 at 21 09 22" src="https://github.com/dart-lang/linter/assets/20165741/d8afa2d0-728c-44fa-a81d-27331fac7161">


It also does not happen if the `///` is on a different line:

<img width="205" alt="Screenshot 2023-12-28 at 21 10 28" src="https://github.com/dart-lang/linter/assets/20165741/135e1d6b-a355-455e-8090-b88a0b44b465">

**Comment by DanTup** [comment]
I can reproduce this - you need to have `dart.showTodos` enabled in VS Code (which is the default, but I suspect is commonly turned off by people working on large multi-developer projects!):

![image](https://github.com/dart-lang/linter/assets/1078012/ff22afb5-997b-40be-a102-2faf8d5e3dbb)

There's some logic in the server to support wrapping of TODOs if the next line is indented further:

![image](https://github.com/dart-lang/linter/assets/1078012/9662b970-7d6b-4d36-82d7-9f0687c22fd6)

https://github.com/dart-lang/sdk/blob/d1717197ec65fd5f207b56252116ba5f90116be5/pkg/analyzer/lib/src/error/todo_finder.dart#L95

It's probably not considering the next comment starting with `///` but should.

**Comment by rrousselGit** [comment]
TIL that you can put an extra spacing for multiline TODOs. Good to know!
